### ISSUE TITLE
Duplicated groups in group hierarchy (when data contains both skos:member _and_ isothes:subGroup)

### ISSUE THREAD

**Issue by tfrancart** [NatLibFi/Skosmos#385]
Hello

I have duplicated groups in the group hierarchy :

![duplicated-groups](https://cloud.githubusercontent.com/assets/2728945/11741427/50485334-9ff9-11e5-80f0-001c16e60269.png)

The data I am working on uses both skos:member and isothes:subGroup to create the group hierarchy, this may be the cause of the problem.
Similarly, groups are declared with both a skos:Collection type, an isothes:ConceptGroup type, and possibly a third class - subclass of isothes:ConceptGroup :

![both-skos_member_and_isothes_subgroup](https://cloud.githubusercontent.com/assets/2728945/11741509/d85df120-9ff9-11e5-9dcf-1ba226ad806e.png)

A DISTINCT is probably needed in the query that retrieves subgroups.

I am using v1.3-maintenance branch.

**Comment by tfrancart** [comment]
Problem lies in listConceptGroups methods, the ?children variable contains duplicated children because it relies on both skos:member and isothes:subGroup :

```
SELECT ?group (GROUP_CONCAT(STR(?child)) as ?children) ?label ?members ?notation
WHERE {
 $gc {
   ?group a <$groupClass> .
   OPTIONAL { ?group skos:member|isothes:subGroup ?child .
              ?child a <$groupClass> }
   BIND(EXISTS{?group skos:member ?submembers} as ?members)
   OPTIONAL { ?group skos:prefLabel ?label }
   OPTIONAL { ?group rdfs:label ?label }
   FILTER (langMatches(lang(?label), '$lang'))
   OPTIONAL { ?group skos:notation ?notation }
 }
}
GROUP BY ?group ?label ?members ?notation
ORDER BY lcase(?label)
```

Could the property to be used be externalised in the vocabularies config file, just like the group class is ?

**Comment by osma** [comment]
Does it help if you change the expression to `(GROUP_CONCAT(DISTINCT STR(?child)) as ?children)` ?

**Comment by tfrancart** [comment]
Indeed, this fixes the problem. Do you think this modification can be included in 1.3-maintenance as well as 1.4 ?

Thanks a lot

**Comment by osma** [comment]
Fixed on both master and v1.3-maintenance.

1.4 will be released soon though, then we will stop maintaining v1.3-maintenance.
